### PR TITLE
feat(grid): passing the dom event through editing events Closes #8354

### LINKED ISSUE #8354: No way to find why editing is over
## Description  
Right now editing is over when `Enter`, `ESC` or `F2` key is pressed, on `sorting`, `filtering`, `searching`, `hiding`, `paging`, `resizing`, `pinning` and `moving` operations as well as when row editing action buttons are pressed. However there is no way for developer to distinguish which cause the editing to end.
For example if the user edits a row, and click on a new row, developer may need to show a message to the user that edited row changes will be committed. This is not possible now. 

 * igniteui-angular version: 10.1
 * browser: all

## Steps to reproduce  

1. In grid with row editing start editing row.
2.1 Click on editable cell in row.
2.2 Click on `done` button.
3. Try to cancel submitting of the change when 2.1 happens but not when 2.2 happens. 

## Result  
There is no way to find what caused end of editing.

## Expected result  
There should be a way to find what caused end of editing. Probably a property could be added to `IGridEditEventArgs` describing the reason which forced the end of editing.

### PR DESCRIPTION
**Pull request by Aleksandyr**
Closes #8354

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/grid-base.directive.ts [hunk 1]
```diff
@@ -6550,22 +6550,22 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     /**
      * @hidden @internal
      */
-    endRowTransaction(commit: boolean, row: IgxRow) {
+    endRowTransaction(commit: boolean, row: IgxRow, event?: Event) {
         row.newData = this.transactions.getAggregatedValue(row.id, true);
         let rowEditArgs = row.createEditEventArgs();
```
Comment (wnvko) [inline projects/igniteui-angular/src/lib/grids/grid-base.directive.ts:6555]: pass `event` here too

File: projects/igniteui-angular/src/lib/grids/grid-base.directive.ts [hunk 2]
```diff
@@ -6650,7 +6650,7 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
                 const showIndex = isInView ? -1 : dataIndex;
                 this.showSnackbarFor(showIndex);
             });
-            cancelable = this.gridAPI.submit_add_value();
+            cancelable = this.gridAPI.submit_add_value(event);
             if (!cancelable) {
                 const args = row.createEditEventArgs();
```
Comment (wnvko) [inline projects/igniteui-angular/src/lib/grids/grid-base.directive.ts:6655]: pass `event` here too

**Review by wnvko — CHANGES_REQUESTED** [review]
There are several additional places where we can pass the `event` argument.
First allow `IgxGridCRUDService.beginAddRow` to accept `event` argument. Then pass it to `createEditEventArgs` where it is called in the function body.
Other places where we can pass `event`:
- `IgxGridNavigationService.handleAlt` pass the event to `IgxGridCRUDService.beginAddRow`.
- `IgxGridCRUDService.beginRowEdit` pass the event to `createEditEventArgs` call.
- `IgxGridCRUDService.beginAddRow` pass the event to `createEditEventArgs` call.
- `IgxGridCRUDService.enterEditMode` pass the event to `IgxGridCRUDService.beginAddRow` call.
- `IgxGridEditingActionsComponent.startEdit` pass the event to -`IgxGridCRUDService.enterEditMode` call.
- `IgxGridExpandableCellComponent.onIndicatorFocus` pass the event to `GridBaseAPIService.submit_value` call.
- `IgxGridCellComponent.pointerdown` pass the event to `GridBaseAPIService.submit_value call`.

**Comment by wnvko** [comment]
Follow the next steps:
1. In dev samples go to Grid Add Row.
2. Add several rows.
3. Put one of the added rows in edit mode.
4. Hover over other row and click add row button in row overlay.
5. Try to determine why the previous row exited edit mode and submitted its data.

Here is what happens:
- `IgxGridEditingActionsComponent.addRowHandler` is not passing the `event` to `IgxGridBaseDirective.beginAddRowByIndex`.
- `IgxGridBaseDirective.beginAddRowByIndex` in `row.onAnimationEnd` handler the event is not passed to `cell.setEditMode`
- this last one does not even accept the `event` argument, but it should. Once it has the `event` argument it should be passed to `IgxGridCRUDService.enterEditMode`.
- in `IgxGridCRUDService.enterEditMode` call to `beginAddRow` should pass the `event`.
- in `IgxGridCRUDService.beginAddRow` should accept `event` and pass it where event args are generated.

**Review by wnvko — CHANGES_REQUESTED** [review]
Follow the next steps:
1. In dev samples go to Grid Add Row.
2. Add several rows.
3. Put one of the added rows in edit mode.
4. Hover over other row and click add row button in row overlay.
5. Try to determine why the previous row exited edit mode and submitted its data.

Here is what happens:
- `IgxGridEditingActionsComponent.addRowHandler` is not passing the `event` to `IgxGridBaseDirective.beginAddRowByIndex`.
- `IgxGridBaseDirective.beginAddRowByIndex` in `row.onAnimationEnd` handler the event is not passed to `cell.setEditMode`
- this last one does not even accept the `event` argument, but it should. Once it has the `event` argument it should be passed to `IgxGridCRUDService.enterEditMode`.
- in `IgxGridCRUDService.enterEditMode` call to `beginAddRow` should pass the `event`.
- in `IgxGridCRUDService.beginAddRow` should accept `event` and pass it where event args are generated.

**Review by wnvko — APPROVED** [review]

**Comment by hanastasov** [comment]
Taking any of the following actions while a cell is in edit mode, will exit edit mode. In the `cellEditExit` event handler and consecutive editing events emitted, the `event` property of the `IGridEditDoneEventArgs` object is undefined:
- resizing a column
- hiding a column through the toolbar or API
- pinning a column through the toolbar or API
- moving a column through the toolbar or API, or drag and drop funct
- paging through the UI or via API
- filtering through in row filtering UI
- ESF filtering actions (filtering, hiding, inning, clear column filter)

**Comment by ddincheva** [comment]
> Taking any of the following actions while a cell is in edit mode, will exit edit mode. In the `cellEditExit` event handler and consecutive editing events emitted, the `event` property of the `IGridEditDoneEventArgs` object is undefined:
> 
> * resizing a column
> * hiding a column through the toolbar or API
> * pinning a column through the toolbar or API
> * moving a column through the toolbar or API, or drag and drop funct
> * paging through the UI or via API
> * filtering through in row filtering UI
> * ESF filtering actions (filtering, hiding, inning, clear column filter)
> 
> IgxGridEditingActionsComponent should be checked too.

In listed scenarios, the editing is ended due to the fact one of these operations is performed, there is no actual DOM event that is triggered the end editing, that is why the event is undefined. Also when the action is triggered from the grid API, again there is no DOM event that can be passed as an argument.

**Pull request merged**